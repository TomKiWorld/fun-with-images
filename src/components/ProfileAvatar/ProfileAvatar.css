.profile-avatars {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  width: 14rem;
  margin: 0 auto;
}

.avatar-img.selected {
  border: 0.1rem solid #E900A3;
}
