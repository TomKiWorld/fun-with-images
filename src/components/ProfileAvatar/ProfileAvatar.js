import React from 'react';
import avatarOne from '../../images/FaceOne.svg';
import avatarTwo from '../../images/FaceTwo.svg';
import avatarThree from '../../images/FaceThree.svg';
import avatarFour from '../../images/FaceFour.svg';

import './ProfileAvatar.css';

const setclasses = (text, avatar) => {
  return text === avatar ? 'avatar-img hyve grow selected' : 'avatar-img hyve grow';
}

const ProfileAvatar = ({ title, avatar, selectAvatar }) => {
  return (
    <div>
      <p className='db fw6 lh-copy f6'>{title}</p>
      <div className='profile-avatars'>
        <img
          className={setclasses('avatarOne', avatar)}
          src={avatarOne}
          alt='Avatar one'
          onClick={() => selectAvatar('avatarOne')}
        />
        <img
          className={setclasses('avatarTwo', avatar)}
          src={avatarTwo} 
          alt='Avatar two'
          onClick={() => selectAvatar('avatarTwo')}
        />
        <img
          className={setclasses('avatarThree', avatar)}
          src={avatarThree} 
          alt='Avatar three' 
          onClick={() => selectAvatar('avatarThree')}
        />
        <img
          className={setclasses('avatarFour', avatar)}
          src={avatarFour} 
          alt='Avatar four' 
          onClick={() => selectAvatar('avatarFour')}
        />
      </div>
    </div>
  );
}

export default ProfileAvatar;
