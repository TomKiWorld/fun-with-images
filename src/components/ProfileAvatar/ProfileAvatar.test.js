import { shallow } from 'enzyme';
import React from 'react';
import ProfileAvatar from './ProfileAvatar';

describe('ProfileAvatar Component', () => {
  const wrapper =
    shallow(
      <ProfileAvatar 
      title={'Hello'} 
      avatar={'avatarOne'}
      />
      );

  it('Expect to render ProfileAvatar Component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
