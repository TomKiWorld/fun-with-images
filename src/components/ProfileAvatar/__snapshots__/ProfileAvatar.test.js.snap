// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileAvatar Component Expect to render ProfileAvatar Component 1`] = `ShallowWrapper {}`;
